src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -56,68 +58,76 @@ export class ReactNativeErrorHandlers implements Integration {
-
-      tracking.disable();
-      tracking.enable({
-        allRejections: true,
-        onUnhandled: (id: string, error: Error) => {
-          if (__DEV__) {
-            promiseRejectionTrackingOptions.onUnhandled(id, error);
-          }
+      if (this._options.patchGlobalPromise) {
+        this._polyfillPromise();
+      }
 
-          getCurrentHub().captureException(error, {
-            data: { id },
-            originalException: error,
-          });
-        },
-        onHandled: (id: string) => {
-          promiseRejectionTrackingOptions.onHandled(id);
-        },
-      });
+      this._attachUnhandledRejectionHandler();
+      this._checkPromiseAndWarn();
     }
   }
   /**
-   * Gets the promise rejection handlers, tries to get React Native's default one but otherwise will default to console.logging unhandled rejections.
+   * Polyfill the global promise instance with one we can be sure that we can attach the tracking to.
+   *
+   * In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
+   * This is due to a version mismatch between promise versions.
+   * Originally we tried a solution where we would have you put a package resolution to ensure the promise instances match. However,
+   * - Using a package resolution requires the you to manually troubleshoot.
+   * - The package resolution fix no longer works with 0.67 on iOS Hermes.
    */
-  private _getPromiseRejectionTrackingOptions(): PromiseRejectionTrackingOptions {
-    return {
-      onUnhandled: (id, rejection = {}) => {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `Possible Unhandled Promise Rejection (id: ${id}):\n${rejection}`
-        );
+  private _polyfillPromise(): void {
+    /* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-var-requires*/
+    const {
+      polyfillGlobal,
+    } = require("react-native/Libraries/Utilities/PolyfillFunctions");
+
+    // Below, we follow the exact way React Native initializes its promise library, and we globally replace it.
+    const Promise = require("promise/setimmediate/es6-extensions");
+
+    // As of RN 0.67 only done and finally are used
+    require("promise/setimmediate/done");
+    require("promise/setimmediate/finally");
+
+    polyfillGlobal("Promise", () => Promise);
+    /* eslint-enable import/no-extraneous-dependencies,@typescript-eslint/no-var-requires */
+  }
+  /**
+   * Attach the unhandled rejection handler
+   */
+  private _attachUnhandledRejectionHandler(): void {
+    const tracking: {
+      disable: () => void;
+      enable: (arg: unknown) => void;
+      // eslint-disable-next-line import/no-extraneous-dependencies,@typescript-eslint/no-var-requires
+    } = require("promise/setimmediate/rejection-tracking");
+
+    const promiseRejectionTrackingOptions = this._getPromiseRejectionTrackingOptions();
+
+    tracking.disable();
+    tracking.enable({
```

Comment:
While we here, can we leave a comment explaining why we have the disable then enable?

Reply:
Hmm, looks like it just sets the properties on `Promise` to null before setting it again anyways in enable. So it's not needed. It's something we've kept here since the start. I'll remove it.